app/controllers/foreman_inventory_upload/uploads_controller.rb

```diff
@@ -11,14 +11,10 @@ def last
 
     def download_file
       filename = ForemanInventoryUpload.facts_archive_name(params[:organization_id])
-      path = Rails.root.join(ForemanInventoryUpload.uploads_folder, filename)
-      unless File.exist? path
-        return throw_flash_error(
-          "Path doesn't exist: #{path}"
-        )
-      end
+      files = Dir["#{ForemanInventoryUpload.uploads_folder}**/#{filename}"]
```

Comment:
We can be a bit more strict:
```suggestion
      files = Dir["{#{ForemanInventoryUpload.uploads_file_path(filename)},#{ForemanInventoryUpload.done_file_path(filename)}}"]
```

This will look only in done and uploads folder.

Reply:
Thanks! updated